executor/cook/executor.py

```diff
@@ -0,0 +1,344 @@
+    """Retrieves the id of the task.
+
+    Parameters
+    ----------
+    :param task: map
+        The task
+
+    Returns
+    -------
+    :return: The id of the task.
+    """
+    return task['task_id']['value']
+
+
+def create_status(task_id, task_state):
+    """Creates a dictionary representing the task status.
+
+    Parameters
+    ----------
+    :param task_id: The task id.
+    :param task_state: The state of the task to report.
+
+    Returns
+    -------
+    :return: a status dictionary that can be sent to the driver.
+    """
+    return {'task_id': {'value': task_id},
+            'state': task_state,
+            'timestamp': time.time()}
+
+
+def update_status(driver, task_id, task_state):
+    """Sends the status using the driver. 
+
+    Parameters
+    ----------
+    :param driver: MesosExecutorDriver
+        The driver to send the status update to.
+    :param task: map
+        The task whose status update to send.
+    :param task_state: string
+        The state of the task which will be sent to the driver.
+
+    Returns
+    -------
+    :return: Nothing.
+    """
+    logging.debug('Updating task {} state to {}'.format(task_id, task_state))
+    status = create_status(task_id, task_state)
+    driver.sendStatusUpdate(status)
+
+
+def send_message(driver, message, max_message_length):
+    """Sends the message, if it is smaller than the max length, using the driver.
+
+    Parameters
+    ----------
+    :param driver: MesosExecutorDriver
+        The driver to send the message to.
+    :param message: object
+        The raw message to send.
+    :param max_message_length: int
+        The allowed max message length after encoding.
+
+    Returns
+    -------
+    :return: whether the message was successfully sent
+    """
+    logging.debug('Sending framework message {}'.format(message))
+    encoded_message = encode_data(str(message).encode('utf8'))
+    if len(encoded_message) < max_message_length:
+        driver.sendFrameworkMessage(encoded_message)
+        return True
+    else:
+        log_message_template = 'Unable to send message {} as it exceeds allowed max length of {}'
+        logging.info(log_message_template.format(message, max_message_length))
+        return False
+
+
+def launch_task(driver, task, stdout_name, stderr_name):
+    """Launches the task using the command available in the data field.
+
+    Parameters
+    ----------
+    :param driver: MesosExecutorDriver
+        The mesos driver to use.
+    :param task: map
+        The task to execute.
+    :param stdout_name: string
+        The file to use to redirect stdout.
+    :param stderr_name: string
+        The file to use to redirect stderr.
+
+    Returns
+    -------
+    :return: The tuple containing the process launched, stdout file, and stderr file.
+    """
+    task_id = get_task_id(task)
+    try:
+        update_status(driver, task_id, cook.TASK_RUNNING)
+
+        command = decode_data(task['data']).decode('utf8')
```

Comment:
Instead of sending the command as a string, consider sending a json map where one of the keys is "command". That way later on we can add other fields if needed without breaking backwards compatibility.